### ISSUE TITLE
with --cb or --cb_explore VW creates multiple hashes for same feature

### ISSUE THREAD

**Issue by marco-rossi29** [VowpalWabbit/vowpal_wabbit#1724]
#### Describe the bug
Multiple hashes are created for the same feature.
May be related to #1566 and #1687 

#### To Reproduce
`data.txt` contains:
```
1:-1.0:0.9 | c
1:-1.0:0.1 | d
```

Running:
```
vw -d data.txt --cb_explore 4 --epsilon 0.05 --audit
```
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = data.txt
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0
        c:**176892**:1:0@0  Constant:**142052**:1:0@0
0
        c:**176893**:1:0@0  Constant:**142053**:1:0@0
0
        c:**176894**:1:0@0  Constant:**142054**:1:0@0
0
        c:**176895**:1:0@0  Constant:**142055**:1:0@0
0
        c:**176888**:1:0@0  Constant:**142048**:1:0@0
0
        c:**176889**:1:0@0  Constant:**142049**:1:0@0
0
        c:**176890**:1:0@0  Constant:**142050**:1:0@0
0
        c:**176891**:1:0@0  Constant:**142051**:1:0@0
-0.013889 -0.013889            1            1.0        1 1:0.962500        2
-0.262749
        Constant:**142052**:1:-0.262749@4.44444     d:**41884**:1:0@0
0
        d:**41885**:1:0@0   Constant:**142053**:1:0@0
0
        d:**41886**:1:0@0   Constant:**142054**:1:0@0
0
        d:**41887**:1:0@0   Constant:**142055**:1:0@0
-0.261537
        Constant:**142048**:1:-0.261537@4.93827     d:**41880**:1:0@0
0
        d:**41881**:1:0@0   Constant:**142049**:1:0@0
0
        d:**41882**:1:0@0   Constant:**142050**:1:0@0
0
        d:**41883**:1:0@0   Constant:**142051**:1:0@0
-0.069444 -0.125000            2            2.0        1 1:0.962500        2

finished run
number of examples per pass = 2
passes used = 1
weighted example sum = 2.000000
weighted label sum = 0.000000
average loss = -0.069444
total feature number = 4

The model after these 2 events has 6 non-zero weights (which doesn't correspond with what --audit printed above):
```
Version 8.6.1
Id 
Min label:-7.63526
Max label:0
bits:18
lda:0
0 ngram:
0 skip:
options: --cb 4 --cb_explore 4 --cb_type dr --csoaa 4 --epsilon 0.0500000007450581
Checksum: 174952304
:0
41880:-0.337228
41884:-0.364464
142048:-0.594971
142052:-0.594901
176888:-0.261537
176892:-0.262749
```

#### Environment
Both linux and windows. Both with cmd line and python bindings.
Versions: bug was in latest master, and it was there since at least 8.4.0

**Comment by jackgerrits** [comment]
I’ll look into this

**Comment by jackgerrits** [comment]
This is expected behavior, closing
